#  <p style="text-align: center;">Практическое задание к Семинару 4.<br><br>Урок 4. Непрерывная случайная величина.</p>

In [1]:
from math import factorial, exp
import numpy as np
from statistics import NormalDist

**Равномерное распределение.**<br>
***Дисперсия:***
<div class="alert alert-block alert-warning"><font size=4 color=black><b>$D=\frac{\displaystyle (b - a)^2}{12}$ (1)</b></font></div>

***Математическое ожидание:***
<div class="alert alert-block alert-warning"><font size=4 color=black><b>$M(X)=\frac{a + b}{12}$ (2)</b></font></div>

---

**Нормальное распределение:**<br>  
***Функция плотности нормального распределения:*** 
<div class="alert alert-block alert-warning"><font size=4 color=black><b>$\displaystyle f(x)=\frac{1}{\sigma\sqrt{2\pi}}\cdot e^{-\frac{(x-a)^2}{2\sigma^2}}$ (3)</b></font></div>

***Формула расчета $Z$-оценки:***
<div class="alert alert-block alert-warning"><font size=4 color=black><b>$\displaystyle Z=\frac{X-\bar x}{\sigma}$ (3)</b></font></div>   

Таблица 1. Z Таблица (положительные значения)

![Positive Z Table](https://datascience.eu/wp-content/uploads/2020/05/image1.png)

Таблица 2. Z Таблица (отрицательные значения)

![Negative Z Table](https://www.ztable.net/wp-content/uploads/2018/11/negativeztable.png)

In [2]:
def D_rav (a,b):  # Дисперсия
    return ((b-a)**2)/12

In [3]:
def Mean_rav (a,b):  # Математическое ожидание
    return (a + b)/2

In [4]:
def z_value(x,mu,sg): # Приводим к нормальному распределению
    return (x-mu)/sg

## ЗАДАНИЕ 1. 
**Случайная непрерывная величина A имеет равномерное распределение на промежутке (200, 800]. Найдите ее среднее значение и дисперсию.**

**<p style="text-align: center;">
    <font size=10 color=#32CD32><b>-------------------------------</b></font><br>
    <font size=4.5 color=#3ac444>РЕШЕНИЕ:</font><br>
<font size=10 color=#32CD32><b>-------------------------------</b></font></p>**

In [5]:
# Дано
a = 200
b = 800

По формуле (1) находим Среднее значение:

In [6]:
D_rav (a,b)

30000.0

По формуле (2) находим Дисперсию:

In [7]:
Mean_rav (a,b)

500.0

<div class="alert alert-block alert-success"><font size=3 color='b'><b>ОТВЕТ: </b><br>Случайная непрерывная величина A на промежутке (200, 800] равномерного распределения,имеет значение:<br>Cреднее значение: 500.00<br>Дисперсию: 30000.00

---

## ЗАДАНИЕ 2. 
**О случайной непрерывной равномерно распределенной величине B известно, что ее дисперсия равна 0.2. Можно ли найти правую границу величины B и ее среднее значение зная, что левая граница равна 0.5? Если да, найдите ее.**

**<p style="text-align: center;">
    <font size=10 color=#32CD32><b>-------------------------------</b></font><br>
    <font size=4.5 color=#3ac444>РЕШЕНИЕ:</font><br>
<font size=10 color=#32CD32><b>-------------------------------</b></font></p>**

In [8]:
# Дано
D_2 = 0.2
a_2 = 0.5

Используя формулу (1), найдем значание правой границы $b$ величины $B$:<br>$\displaystyle \frac{(b-0.5)^2}{12}=0.2\\b-0.5=\sqrt{2.4}\\b=0.5+\sqrt{2.4}$

In [9]:
b_2 = a_2 + 2.4**0.5
print(f'Правая граница величины B равна: {b_2: .4f}')

Правая граница величины B равна:  2.0492


Используя формулу (2), найдем среднее значение величины:

In [10]:
M_2 = (a_2 + b_2) / 2
print (f'Среднее значение В на промежутке (0.5; {b_2: .4f}), равно M(B) = {M_2: .4f}')

Среднее значение В на промежутке (0.5;  2.0492), равно M(B) =  1.2746


<div class="alert alert-block alert-success"><font size=3 color='b'><b>ОТВЕТ: </b><br>Можно найти правую границу величины B и ее среднее значение!<br>Правая граница распределения величины B: $b\approx2.0492$<br>Среднее значение В на промежутке (0.5; 2.0492), равно M(B) $\approx1.2746$</font></div>

---

## ЗАДАНИЕ 3. 
**Непрерывная случайная величина X распределена нормально и задана плотностью распределения: $\displaystyle f(x)=\frac{1}{4\cdot\sqrt{2\pi}}\cdot e^{-\frac{(x+2)^2}{32}}$, Найдите:**  
   **а) $M(X)$**  
   **б) $D(X)$**  
   **в) $\displaystyle\sigma$ (среднее квадратичное отклонение)**

**<p style="text-align: center;">
    <font size=10 color=#32CD32><b>-------------------------------</b></font><br>
    <font size=4.5 color=#3ac444>РЕШЕНИЕ:</font><br>
<font size=10 color=#32CD32><b>-------------------------------</b></font></p>**

Из формулы (3), плотности нормального распределения найдём:<br>$M(X)=a=-2\\\sigma = 4\\D(X)=\sigma^2=16$

<div class="alert alert-block alert-success"><font size=3 color='b'><b>ОТВЕТ: </b><br>$M(X)=-2\\D(X)=16\\\sigma = 4$</font></div>

---

## ЗАДАНИЕ 4. 
**Рост взрослого населения города X имеет нормальное распределение. Причем, средний рост равен 174 см, а среднее квадратичное отклонение равно 8 см.Какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:**<br>
**а) больше 182 см<br>б) больше 190 см<br>в) от 166 см до 190 см<br>г) от 166 см до 182 см<br>д) от 158 см до 190 см<br>е) не выше 150 см или не ниже 190 см<br>ё) не выше 150 см или не ниже 198 см<br>ж) ниже 166 см.<br><br>**

**<p style="text-align: center;">
    <font size=10 color=#32CD32><b>-------------------------------</b></font><br>
    <font size=4.5 color=#3ac444>РЕШЕНИЕ:</font><br>
<font size=10 color=#32CD32><b>-------------------------------</b></font></p>**

In [11]:
# Дано
mu = 174
sg = 8

**a) больше 182 см.** Определим $Z$-значениe:

In [12]:
x = 182
z_value (x,mu,sg)

1.0

По таблице 1, находим значение Z, которое равно **0.84134**

Учитывая, что нам надо найти $P(X\ge182)$, поэтому вычтем полученное значение из $1$:

In [13]:
P_a = 1 - 0.84134
round (P_a, 4)

0.1587

***Проверим полученное значение с помощью метода cdf, объекта NormalDist стандартного модуля statistics***

In [14]:
P_a_1 = 1- NormalDist().cdf(z_value(x,mu,sg))
round (P_a_1, 4)

0.1587

**$P(X\ge182)=0.1587$**

**б) больше 190 см.** Определим $Z$-значениe:

In [15]:
x_b = 190
z_value (x_b,mu,sg)

2.0

По таблице 1, находим значение Z, которое равно **0.97725**

Учитывая, что нам надо найти $P(X\ge190)$, поэтому вычтем полученное значение из $1$:

In [16]:
P_b = 1 - 0.97725
round (P_b, 4)

0.0228

***Проверим полученное значение с помощью метода cdf, объекта NormalDist стандартного модуля statistics***

In [17]:
P_b_1 = 1- NormalDist().cdf(z_value(x_b,mu,sg))
round (P_b_1, 4)

0.0228

**$P(X\ge190)=0.0228$**

**в) от 166 см до 190 см.** Определим $Z$-значения для обоих показателей роста:

In [18]:
x_v_1 = 166
z_value (x_v_1,mu,sg)

-1.0

По таблице 2, находим значение Z, которое равно **0.15866**

In [19]:
x_v_2 = 190
z_value (x_v_2,mu,sg)

2.0

По таблице 1, находим значение Z, которое равно **0.97725**

Найдем требуемый промежуток:

In [20]:
x_v_P = 0.97725 - 0.15866
round (x_v_P, 4)

0.8186

***Проверим полученное значение с помощью метода cdf, объекта NormalDist стандартного модуля statistics***

In [21]:
P_v = NormalDist().cdf(z_value (x_v_2,mu,sg))-NormalDist().cdf(z_value (x_v_1,mu,sg))
round (P_v,4)

0.8186

$P(166\le X\le 190)=0.8186$  

**г) от 166 см до 182 см.** Определим $Z$-значения для обоих показателей роста:

In [22]:
x_g_1 = 166 
z_g_1 = 0.15866 # известно из задание в).

In [23]:
x_g_2 = 182
z_value (x_g_2,mu,sg)

1.0

По таблице 1, находим значение Z, которое равно **0.84134**

Найдем требуемый промежуток:

In [24]:
x_g_P = 0.84134 - 0.15866
round (x_g_P, 4)

0.6827

***Проверим полученное значение с помощью метода cdf, объекта NormalDist стандартного модуля statistics***

In [25]:
P_v = NormalDist().cdf(z_value (x_g_2,mu,sg))-NormalDist().cdf(z_value (x_g_1,mu,sg))
round (P_v,4)

0.6827

$P(166\le X\le 182)=0.6827$  

**д) от 158 см до 190 см.** Определим $Z$-значения для обоих показателей роста:

In [26]:
x_d_1 = 158 
z_value (x_d_1,mu,sg)

-2.0

По таблице 2, находим значение Z, которое равно **0.02275**

In [27]:
x_d_2 = 190 
z_d_1 = 0.97725 # известно из задание в).

Найдем требуемый промежуток:

In [28]:
x_d_P = 0.97725 - 0.02275
round (x_d_P, 4)

0.9545

***Проверим полученное значение с помощью метода cdf, объекта NormalDist стандартного модуля statistics***

In [29]:
P_d = NormalDist().cdf(z_value (x_d_2,mu,sg))-NormalDist().cdf(z_value (x_d_1,mu,sg))
round (P_d,4)

0.9545

$P(158\le X\le 190)=0.9545$  

**е) не выше 150 см или не ниже 190 см.** Определим $Z$-значения для обоих показателей роста:

In [30]:
x_e_1 = 150 
z_value (x_e_1,mu,sg)

-3.0

По таблице 2, находим значение Z, которое равно **0.00135**

In [31]:
x_e_2 = 190 
z_e_2 = 1 - 0.97725 # известно из задание в). Т.к. нам надо найти все значения не ниже 190, поэтому вычитаем из единицы
round (z_e_2,6)

0.02275

Найдем требуемый промежуток:

In [32]:
x_e_P = 0.00135 + 0.0228
round (x_e_P, 6)

0.02415

***Проверим полученное значение с помощью метода cdf, объекта NormalDist стандартного модуля statistics***

In [33]:
P_e = (1- NormalDist().cdf(z_value (x_e_2,mu,sg)))+NormalDist().cdf(z_value (x_e_1,mu,sg))
round (P_e,6)

0.0241

$P(X\le150\cup X\ge 190)=0.0241$  

**ё) не выше 150 см или не ниже 198 см.** Определим $Z$-значения для обоих показателей роста:

In [34]:
x_ei_1 = 150 
x_Z_1 = 0.00135 # известно из задания е)

In [35]:
x_ei_2 = 198 
z_value (x_ei_2,mu,sg)

3.0

По таблице 1, находим значение Z, которое равно **0.99865**, т.к. нам надо найти все значения не ниже 198, поэтому вычитаем из единицы полученное значение Z:

In [36]:
round (1 - 0.99865,6)

0.00135

Найдем требуемый промежуток:

In [37]:
x_ei_P = 0.00135 + 0.00135
round (x_ei_P, 6)

0.0027

***Проверим полученное значение с помощью метода cdf, объекта NormalDist стандартного модуля statistics***

In [38]:
P_ei = (1- NormalDist().cdf(z_value (x_ei_2,mu,sg)))+NormalDist().cdf(z_value (x_ei_1,mu,sg))
round (P_ei,6)

0.0027

**$P(X\le150\cup X\ge 198)=0.0027$**

**ж) ниже 166 см.** Определим $Z$-значение:

In [39]:
x_g_1 = 166
z_value (x_g_1,mu,sg)

-1.0

По таблице 2, находим значение Z, которое равно **0.15866**

***Проверим полученное значение с помощью метода cdf, объекта NormalDist стандартного модуля statistics***

In [40]:
P_g = NormalDist().cdf(z_value (x_g_1,mu,sg))
round (P_g,5)

0.15866

$P(X\le166)=0.15866$  

<div class="alert alert-block alert-success"><font size=3 color='b'><b>ОТВЕТ: </b><br> а) $P(X\ge182)\approx0.1587$; <br> 
   б) $P(X\ge190)\approx0.0228$;  <br>
   в) $P(166\le X\le 190)\approx0.8186$;  <br>
   г) $P(166\le X\le 182)\approx0.6827$; <br> 
   д) $P(158\le X\le 190)\approx0.9545$;  <br>
   е) $P(X\le150\cup X\ge 190)\approx0.0241$;  <br>
   ё) $P(X\le150\cup X\ge 198)\approx0.0027$;  <br>
   ж) $P(X\le166)\approx0.15866$;  </font></div>

---

## ЗАДАНИЕ 5. 
**На сколько сигм (средних квадратичных отклонений) отклоняется рост человека, равный 190 см, от математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см?.<br><br>**

**<p style="text-align: center;">
    <font size=10 color=#32CD32><b>-------------------------------</b></font><br>
    <font size=4.5 color=#3ac444>РЕШЕНИЕ:</font><br>
<font size=10 color=#32CD32><b>-------------------------------</b></font></p>**

Воспользуемся формулой (3):

$\displaystyle Z=\frac{X-\bar x}{\sigma}=\frac{190-178}{\sqrt{25}}=\frac{12}{5}$    

In [41]:
Z=(190-178)/25**(1/2)
Z   

2.4

<div class="alert alert-block alert-success"><font size=3 color='b'><b>ОТВЕТ: </b><br> Рост человека, равный 190 см, отклоняется от математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см на <b>2,4 сигмы.</b>  </font></div>